src/lib/utils/balancer/lido.ts

```diff
@@ -47,15 +55,23 @@ export function getWstETHByStETH(stETHAmount: BigNumberish) {
 }
 
 /**
- * @notice Get amount of stETH for a given amount of wstETH
+ * @notice Get amount of stETH for a given amount of wstETH or vice versa
  */
-export function getStETHByWstETH(wstETHAmount: BigNumberish) {
-  const wstETH = new Contract(
-    wstEthAddress,
-    [
-      'function getStETHByWstETH(uint256 wstETHAmount) external view returns (uint256)',
-    ],
+export async function getStETHOrWstETH(
+  amount: BigNumber,
+  network: Network,
+  isStEthByWstEth = true
+) {
+  const wstEthRateProvider = new Contract(
+    wstEthRateProvidersMap[network],
+    ['function getRate() external view returns (uint256)'],
     rpcProviderService.jsonProvider
   );
-  return wstETH.getStETHByWstETH(wstETHAmount);
+
+  const rate = await wstEthRateProvider.getRate();
+  amount = isStEthByWstEth
+    ? amount.mul(rate).div(ONE)
+    : amount.mul(ONE).div(rate);
+
+  return amount;
```

Comment:
Let's refactor this function a bit.
1. Let's rename it to be a bit more clear.
2. Since it should only be relevant on mainnet lets throw if not.
3. Let's change the inputs to an object so that it's clear externally what the direction of the conversion is.

```ts
const MAINNET_RATE_PROVIDER_ADDRESS = '0x72d07d7dca67b8a406ad1ec34ce969c90bfee768'

type ConversionParams = {
  amount: BigNumber,
  isWrap: boolean,  // e.g. is stETH to wstETH
}

/**
 * Convert stETH amount to wstETH or vice versa. Only relevant on mainnet when wrapping or unwrapping.
 *
 * @param {BigNumber} amount - The amount to convert, could be stETH or wstETH value.
 * @param {boolean} isWrap - True if wrapping stETH to wstETH, false if unwrapping wstETH to stETH.
 * @returns Converted value for wrap or unwrap, if input is stETH, returns wstETH value and vice versa.
 */
export function convertStEthWrap({ amount, isWrap }: ConversionParams): BigNumber {
  try {
      const rateProvider = new Contract(
        MAINNET_RATE_PROVIDER_ADDRESS,
        ['function getRate() external view returns (uint256)'],
        rpcProviderService.jsonProvider
      );

      const rate = await rateProvider.getRate();

      return isWrap
        ? amount.mul(rate).div(ONE)
        : amount.mul(ONE).div(rate);
   } catch (error) {
      throw new Error(`Failed to convert stETH: ${error}`);
   }
}
```